### DISCUSSION TITLE
Forgot whether the current mouse is in English or Chinese input state during use?

### DISCUSSION THREAD

**Discussion by Prague2049** [meow-edit/meow#490] [General]
Is Meow possible to identify whether the mouse is currently in Chinese or text input state?
Forgot whether the current mouse is in English or Chinese input state during use? 
In the keypad mode, the result of combining keys is different in Chinese and English input states.

**Comment by DogLooksGood** [comment]
Cloud you give an example? Do you mean using meow with system input method?

**Comment by Prague2049** [comment]
For example:Currently I have set up the mouse to display in two modes as follows.
(setq meow-cursor-type-insert '(hollow . 4))
(setq meow-cursor-type-keypad '(bar . 1)) 
<img width="43" alt="image" src="https://github.com/meow-edit/meow/assets/86907530/f84876f8-bdbf-4252-99cf-eb4e97fee712">

In keypad mode: make the mouse bar color blue when the input status is English. When the input state is Chinese, the color of the mouse bar is red or can be defined as other colors. This way I can flexibly switch the input method according to the displayed color.

<img width="39" alt="image" src="https://github.com/meow-edit/meow/assets/86907530/d574d828-b2cf-40dc-bd13-8fc6ab88b63c">

**Reply by DogLooksGood** [reply]
If you are using the system input method, AFAIK, there's no easy way to detect the current input method.

**Reply by Prague2049** [reply]
I use rime in macOS.
